nexus/db-queries/src/db/datastore/trust_quorum.rs

```diff
@@ -65,39 +65,55 @@ fn epoch_to_i64(epoch: Epoch) -> Result<i64, Error> {
 }
 
 impl DataStore {
-    /// Return all `HwBaseboardId`s for a given rack that has run LRTQ
+    /// Return a `RackUuid` for each trust quorum along with its latest `Epoch`.
     ///
-    /// No need for pagination, as there at most 32 member sleds per rack
-    pub async fn lrtq_members(
+    /// For now, since we do not have multirack, and we aren't sure how big
+    /// those clusters are going to be we return all values and don't bother
+    /// paginating. The current `SQL_BATCH_SIZE` is also 1000, and it's unlikely
+    /// that there will ever be more than 1000 racks in a single fleet, sharing
+    /// a single CRDB cluster.
+    pub async fn tq_get_all_rack_id_and_latest_epoch(
         &self,
         opctx: &OpContext,
-        rack_id: RackUuid,
-    ) -> ListResultVec<HwBaseboardId> {
+    ) -> ListResultVec<(RackUuid, Epoch)> {
         opctx.authorize(authz::Action::ListChildren, &authz::FLEET).await?;
         let conn = &*self.pool_connection_authorized(opctx).await?;
 
-        use nexus_db_schema::schema::hw_baseboard_id::dsl as hw_baseboard_id_dsl;
-        use nexus_db_schema::schema::lrtq_member::dsl as lrtq_member_dsl;
+        use nexus_db_schema::schema::trust_quorum_configuration::dsl;
 
-        lrtq_member_dsl::lrtq_member
-            .filter(lrtq_member_dsl::rack_id.eq(rack_id.into_untyped_uuid()))
-            .inner_join(hw_baseboard_id_dsl::hw_baseboard_id.on(
-                hw_baseboard_id_dsl::id.eq(lrtq_member_dsl::hw_baseboard_id),
-            ))
-            .select(HwBaseboardId::as_select())
-            .load_async(conn)
-            .await
-            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+        let values: Vec<(DbTypedUuid<RackKind>, i64)> =
+            dsl::trust_quorum_configuration
+                .select((dsl::rack_id, dsl::epoch))
+                .order_by((dsl::rack_id, dsl::epoch.desc()))
+                .distinct_on(dsl::rack_id)
```

Comment:
This doesn't look like it has a `WHERE` clause, right? I'm surprised it passes our `disallow_full_table_scans` setting.